plugins/scaffolder-backend/src/scaffolder/stages/prepare/azure.ts

```diff
@@ -22,24 +22,47 @@ import { InputError, Git } from '@backstage/backend-common';
 import { PreparerBase, PreparerOptions } from './types';
 import parseGitUrl from 'git-url-parse';
 import { Config } from '@backstage/config';
+import { Logger } from 'winston';
+import {
+  readAzureIntegrationConfigs,
+  AzureIntegrationConfig,
+} from '@backstage/integration';
 
 export class AzurePreparer implements PreparerBase {
-  private readonly privateToken: string;
+  private readonly integrations: AzureIntegrationConfig[];
+  private readonly scaffolderToken: string | undefined;
 
-  constructor(config: Config) {
-    this.privateToken =
-      config.getOptionalString('scaffolder.azure.api.token') ?? '';
+  constructor(config: Config, { logger }: { logger: Logger }) {
+    this.integrations = readAzureIntegrationConfigs(
+      config.getOptionalConfigArray('integrations.azure') ?? [],
+    );
+
+    if (!this.integrations.length) {
+      logger.warn(
+        'Integrations for Azure in Scaffolder are not set. This will cause errors in a future release. Please migrate to using integrations config and specifying tokens under hostnames',
```

Comment:
This would spam ppl quite a bit won't it? It would be completely valid to not have an azure config. But then again this never actually happens either right, since `readAzureIntegrationConfigs` always returns at least one entry.

Same with the other cases below